src/main/kotlin/app/batch/HBaseWriter.kt

```diff
@@ -28,41 +29,41 @@ class HBaseWriter(private val connection: Connection) : ItemWriter<DecompressedS
     override fun write(items: MutableList<out DecompressedStream>) {
         items.forEach {
             val fileName = it.fileName
-            val dataKeyResult: DataKeyResult = getDataKey(fileName)
-            val reader = BufferedReader(InputStreamReader(it.inputStream))
-            var line: String? = null
-            var id: String?
-            while ({ line = reader.readLine(); line }() != null) {
-                try {
-                    val parser: Parser = Parser.default()
-                    val stringBuilder = StringBuilder(line)
-                    val json = parser.parse(stringBuilder) as JsonObject
-                    id = getId(json, fileName)?.toJsonString()
-                    logger.info("Parsing DB object of id $id  in the file $fileName")
+            val filenamePattern = """(?<database>[a-z-]+)\.(?<collection>[a-z-]+)\.\d+\.json\.gz\.enc$"""
+            val filenameRegex = Regex(filenamePattern, RegexOption.IGNORE_CASE)
+            val matchResult = filenameRegex.find(fileName)
+            if (matchResult != null) {
+                val groups = matchResult.groups
+                val database = groups[1]!!.value // can assert nun-null as it matched on the regex
+                val collection = groups[2]!!.value // ditto
+                val dataKeyResult: DataKeyResult = getDataKey(fileName)
+                val reader = BufferedReader(InputStreamReader(it.inputStream))
+                var line: String? = null
+                while ({ line = reader.readLine(); line }() != null) {
+                    try {
+                        val parser: Parser = Parser.default()
+                        val stringBuilder = StringBuilder(line)
+                        val json = parser.parse(stringBuilder) as JsonObject
+                        val id = json.obj("_id")?.toJsonString()
+                        if (StringUtils.isNotBlank(id)) {
+                            val encryptionResult = encryptDbObject(dataKeyResult, line!!, fileName, id)
+                            logger.info("result: '$encryptionResult', fileName: '$fileName'.")
+                            val message = MessageProducer().produceMessage(json, encryptionResult, dataKeyResult,
+                                                                            database, collection)
+                            logger.info("Message: '$message'.")
+                        }
+                    }
+                    catch (e: Exception) {
+                        logger.error("Error while parsing record from '$fileName': '${e.message}'.", e)
```

Comment:
Is it possible the error message contains message data? I.e if it doesn't parse, it might log out the specific bit where the parsing failed?

Reply:
Ditto - same concern.

Reply:
Removed the exception message.